src/diff/component.js

```diff
@@ -142,6 +142,19 @@ export function renderComponent(
 
 	if ((tmp = options._render)) tmp(newVNode);
 
+	// Root nodes signal that we attempt to render into a specific DOM node
+	// on the page. Root nodes can occur anywhere in the tree and not just
+	// at the top.
+	if (newProps._parentNode) {
+		parentDom = newProps._parentNode;
+
```

Comment:
Adding the following before checking `startDom` fixes the skipped tests for me. I think what it is saying is "Let's resume the diff of the portal from the previously rendered child of the portal"

```suggestion

		if (oldVNode && oldVNode._dom) {
			startDom = oldVNode._dom;
		}

```

Reply:
I think we also need to capture the previous value of `startDom` here and return it at the end of this function so that the diff can resume from where it left off before the Portal. In other words, the diff should act as if the portal isn't there and so we need to manage the `startDom` pointer so that VNodes around the portal see no difference (i.e. startDom doesn't change).

As this is implemented today, this function will return the `nextSibling` of the last child of the Portal's children which won't be in the same DOM tree as the rest of the diff.

I don't think our tests hit this today cuz most of them put portals at the end of a `<div>`:
```
<div>
    <p>Hello</p>
    {createPortal()}
</div>
```
at which point the `nextSibling` value of the Portal child is thrown away since it is the last child of a div.

I'm working on a test case for this bug...

Reply:
Okay here is the test case that fails:
```js
it("should not needlessly append siblings to Portals", () => {
  /** @type {() => void} */
  let update;
  function Foo(props) {
    const [state, setState] = useState(false);
    update = () => setState((state) => !state);
    return (
      <div>
        <p>Hello</p>
        {createPortal(props.children, scratch)}
        <p>World!</p>
      </div>
    );
  }

  render(
    <Foo>
      <div>foobar</div>
    </Foo>,
    scratch
  );
  expect(scratch.innerHTML).to.equal(
    "<div>foobar</div><div><p>Hello</p><p>World!</p></div>"
  );

  clearLog();
  update();
  rerender();

  expect(scratch.innerHTML).to.equal(
    "<div>foobar</div><div><p>Hello</p><p>World!</p></div>"
  );
  expect(getLog()).to.deep.equal([]);
});
```

Reply:
That's amazing! You are right on the money with this. Storing the previous `startDom` pointer and returning it at the end is what makes all tests pass 👍